### DISCUSSION TITLE
feature request: ch-image inspect

### DISCUSSION THREAD

**Discussion by rarensu** [hpc/charliecloud#1562] [General]
Docker, Podman, and Singularity all have inspect features with varying utility. The most important use of these in my experience has been to extract the ENTRYPOINT of a container image when the original Dockerfile is not available. Because entrypoint is not a supported feature in charliecloud, in order to actually use some images, I have to inspect the container image with podman or singularity to get a copy of the entrypoint and paste it into a regular executable file. For example, NVIDIA container registry publishes a number of images that we really use all the time in production, such as LAMMPS, which has an entrypoint. Unfortunately, the source for the nvidia registry is not open. I do not know any way to view NVIDIA's original Dockerfiles. Probably, this is intentional. I wouldn't be surprised if there are trade secrets in there. 

It doesn't really make sense to install Podman just to get access to Podman inspect, if I am going to continue to use ch-image for build operations anyways. It would be nice if there was a ch-image inspect that could give me the information I needed. I suppose it would also be acceptable if charliecloud just supported the ENTRYPOINT feature. I don't know which is easier to implement.

**Comment by reidpr** [comment]
We do have an issue for `ENTRYPOINT`, #780. I recall some discussion more recently than the latest comment on that issue, but it seems to have not made its way into the issue.

It sounds like your top priority is just to be able to *see* the `ENTRYPOINT`, and a second priority would be a way to execute it?

**Reply by rarensu** [reply]
Yes, that is accurate.
